src/main/java/de/unistuttgart/regexgamebackend/data/Configuration.java

```diff
@@ -0,0 +1,58 @@
+package de.unistuttgart.regexgamebackend.data;
+
+import java.util.EnumSet;
+import java.util.UUID;
+import javax.persistence.*;
+import javax.validation.Valid;
+import javax.validation.constraints.Size;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.hibernate.validator.constraints.Range;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * Contains all data that have to be stored to configure a regex game.
+ */
+@Entity
+@Data
+@NoArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@Validated
+public class Configuration {
+
+    /**
+     * A unique identifier for the configuration.
+     */
+    @Id
+    @GeneratedValue(generator = "uuid")
+    UUID id;
+
+    /**
+     * Allowed regex structures for this game.
+     */
+    @Size(min = 1)
+    EnumSet<RegexStructure> allowRegexStructures;
+
+    /**
+     * Game complexity factor.
+     */
+    @Range(min = 1)
+    @Valid
+    double complexity = 1;
```

Comment:
Wait, a double `>= 1`?
What's the scale then?
I would have either expected `0 <= x <= 1`, or a long `x >= 1`.
Either that, or you thoroughly need to document what you expect.

Reply:
This is an arbitrary number of no real scale handled in the frontend.
It is probabily easier to just always start with a simple game and always increase complexity gradually.
I will remove this configuration option.